semanticz-connector-neptune/src/test/java/zone/cogni/semanticz/connectors/neptune/NeptuneRdfStoreServiceTest.java

```diff
@@ -0,0 +1,244 @@
+package zone.cogni.semanticz.connectors.neptune;
+
+import org.apache.jena.query.Query;
+import org.apache.jena.query.QueryExecution;
+import org.apache.jena.query.QueryFactory;
+import org.apache.jena.query.ResultSet;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdfconnection.RDFConnection;
+import org.apache.jena.rdfconnection.RDFConnectionRemote;
+import org.apache.jena.rdfconnection.RDFConnectionRemoteBuilder;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.MockedStatic;
+import org.mockito.junit.jupiter.MockitoExtension;
+import zone.cogni.sem.jena.template.JenaResultSetHandler;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class NeptuneRdfStoreServiceTest {
+
+    @Mock
+    private RDFConnectionRemoteBuilder mockBuilder;
+    @Mock
+    private RDFConnection mockConnection;
+    @Mock
+    private QueryExecution mockQueryExecution;
+    @Mock
+    private Model mockModel;
+    @Mock
+    private ResultSet mockResultSet;
+
+    private NeptuneRdfStoreService rdfStoreService;
+
+    @BeforeEach
+    void setUp() {
+        rdfStoreService = new NeptuneRdfStoreService("http://localhost:8182");
```

Comment:
Would it be possible to parameterize this (e.g. by a system var) 
- currently we run the tests on Jenkins for GraphDB / Virtuoso (if the servers are available), producing a report: https://jenkins.cogni.zone/job/cz/job/semanticz/job/connectors/job/builds/job/semanticz-connectors/Virtusoso_20test_20report/

This said, I am not sure we are able to run the tests for Neptune as I doubt there is any Neptune around to be used for testing, but will check with sysadmins.